private/version/checker/service.go

```diff
@@ -85,31 +86,42 @@ func (srv *Service) Run(ctx context.Context) (err error) {
 }
 
 // IsAllowed returns whether if the Service is allowed to operate or not
-func (srv *Service) IsAllowed(ctx context.Context) bool {
+func (srv *Service) IsAllowed(ctx context.Context) (version.OldSemVer, bool) {
```

Comment:
We should be using the new version struct, as the old should not be used going forward.